_source/_posts/2020-12-16-react-login.md

```diff
@@ -0,0 +1,599 @@
+
+```jsx
+import React from 'react';
+import { Link } from 'react-router-dom';
+
+function Header() {
+  return (
+    <header>
+      <div>React Login</div>
+      <ul className="menu">
+        <li><Link to="/">Home</Link></li>
+        <li><Link to="/private">Private</Link></li>
+      </ul>
+    </header>
+  );
+}
+
+export default Header;
+```
+
+Next, open the existing `src/App.js`. This is the main component of the application. Edit the contents to match the following code.
+
+```jsx
+import React from 'react';
+import './App.css';
+import Header from './Header';
+import Home from './Home';
+import Private from './Private';
+import { BrowserRouter, Route } from 'react-router-dom';
+
+function App() {
+  return (
+    <div className="App">
+      <div className="page">
+        <div className="content">
+          <BrowserRouter>
+             <Header/>
+             <Route path='/' exact={true} component={Home}/>
+             <Route path='/private' exact={true} component={Private}/>
+          </BrowserRouter>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+export default App;
+```
+
+You can see the `BrowserRouter` component containing the routes for the home page and the private page. It also contains the `Header` component. This component is not a separate route, but it must be contained within the `BrowserRouter` in order for the router links within the header to work. The application still needs some styling. Open `src/App.css` and replace its contents with the CSS code below.
+
+```css
+.App header {
+  background-color: #282c34;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  justify-content: space-between;
+  color: white;
+  padding: 0.5rem 1rem;
+}
+
+ul.menu {
+  list-style: none;
+}
+
+ul.menu li {
+  display: inline;
+  padding: 12px;
+}
+
+ul.menu a {
+  color: #ffffff;
+}
+
+.page {
+  text-align: center;
+}
+
+button {
+  background-color: #6200ee;
+  color: #ffffff;
+  border: none;
+  padding: 12px 16px;
+  text-transform: uppercase;
+  font-weight: bold;
+  cursor: pointer;
+}
+
+button:hover {
+  background-color: #3700b3;
+}
+
+.warning {
+  width: 800px;
+  margin: 24px auto;
+  border: 5px solid #ff0000;
+  border-radius: 15px;
+}
+
+.warning h1 {
+  margin-top: 0;
+  color: white;
+  font-size: 48px;
+  line-height: 96px;
+  background-color: #ff0000;
+  text-transform: uppercase;
+}
+
+.warning h2 {
+  font-size: 36px;
+  text-transform: uppercase;
+}
+```
+
+This completes the basic application. You can test it by running the following command.
+
+```bash
+npm start
+```
+
+This should build the JavaScript code and then automatically open your browser at `http://localhost:3000/`. You will see the home page and when you click the button, you will be taken to the restricted page. At this point, anybody could see the restricted page. No user authentication has been added so far.
```

Comment:
Click on which button?

Reply:
There's only one. It says "Visit Restricted Page" on it.